kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/KafkaStreamsApp.java

```diff
@@ -142,21 +145,31 @@ protected void doInit() {
             System.exit(1);
           });
       this.shutdownDuration = getShutdownDuration();
+      this.startupDelay = getStartupDelay();
       getLogger().info("kafka streams topologies: {}", topology.describe());
     } catch (Exception e) {
       getLogger().error("Error initializing - ", e);
-      e.printStackTrace();
       System.exit(1);
     }
   }
 
   @Override
   protected void doStart() {
     try {
+      isSleeping = true;
```

Comment:
Lets move this also to private function.

Reply:
Done